src/nora/components/KabobMenu/KabobMenuContainer.js

```diff
@@ -36,27 +42,55 @@ export const KabobMenuContainer = ({
     setActiveOption(index)
   }
 
+  const handleOnKeyDown = (ev) => {
+    /**
+     * Edge case:
+     * If we've been passed in a focus ref (focusRefResolved), we're presumably being used from within the
+     * data grid. So, ugly as it may be, we need to temporarily prevent the data grid's keydown listener
+     * from receiving keydown events until the popover options are no longer opened.
+     */
+    if (focusRefResolved && isOpen) {
+      ev.stopPropagation()
+    }
+
+    handlePopoverNavigation({
+      ev: ev,
+      items: items,
+      onChange: handleOnChange,
+      optionsRefs: optionsRefs,
+      activeOption: activeOption,
+      setActiveOption: setActiveOption,
+      scrollItemIntoView: scrollItemIntoView,
+      showPopover: isOpen,
+      setShowPopover: setIsOpen,
+      setSelectedAndActiveOptions: setSelectedAndActiveOptions,
+    })
+  }
+
+  const popoverTrapRef = React.useRef()
```

Comment:
This whole section basically:

- puts focus on the first focusable element in the popover
- adds listener if we've opened the popover
- removes the listener if we tear down the component (if the ref isn't removed already which is usually the case)